src/mixed_arena.h

```diff
@@ -369,21 +369,31 @@ template<typename SubType, typename T> class ArenaVectorBase {
 
   void insertAt(size_t index, T item) {
     assert(index <= size()); // appending is ok
-    resize(size() + 1);
-    for (auto i = size() - 1; i > index; --i) {
-      data[i] = data[i - 1];
+    if (index == size()) {
+      push_back(item); // simple append
+    } else {
+      resize(size() + 1);
+      for (auto i = size() - 1; i > index; --i) {
+        data[i] = data[i - 1];
+      }
+      data[index] = item;
     }
```

Comment:
The previous code was correct, I think? (the loop is not entered when just appending)

If it was correct, I think it's shorter and clearer, personally.

Reply:
Yes, it was correct, but `insertAt` mostly used for external API: https://github.com/WebAssembly/binaryen/search?q=insertAt. And it seems to be called there most often as pash_back, so I think it makes sense to speed up this case

Reply:
Interesting, how much of a speedup did you measure?

(I actually would be surprised if this helps, as you are adding a branch while saving a branch, so it might even out.)

Reply:
Honestly, I didn't measure it. But I just used the same approach that has already been used in one place: https://github.com/WebAssembly/binaryen/pull/4174/files#diff-32b5e1d2d157ef518c21e44e9b85f2047481bf63c5f113c986fdbec7fdee9995L119

Reply:
Good catch that it appears in more than one place.

I think it should not actually matter. As I said, it adds a branch and removes a branch. The bigger issue is the actual allocation - that is unchanged. So unless we see data, I think the clearer code is better. (But maybe others disagree with me on what is clearer, that would be fine of course.)

Reply:
I don't follow. Where is `nextSize = (size + 1) / 2`?

Reply:
Sorry `(size + 1) * 2` not `(size + 1) / 2`:
https://github.com/WebAssembly/binaryen/blob/f532ab50c01b06fe62295e2c72d69d64ce9116c3/src/mixed_arena.h#L220

Reply:
I see, thanks. This does seem like the larger issue.

I'd be that the `*2` approach will be faster. But rather than fix this particular spot in the code, I think we probably want to make `resize()` do the optimal thing itself. Then all callers will be faster.

But, if you want to do this, please measure.

Reply:
Alright I removed this fast path